lua/copilot/panel.lua

```diff
@@ -0,0 +1,500 @@
+
+  return self:get_entry(id), marker_linenr
+end
+
+---@param dir -1|0|1
+function panel:jump(dir)
+  local _, linenr = self:get_entry(dir)
+  if linenr then
+    vim.api.nvim_win_set_cursor(self.winid, { linenr, 0 })
+  end
+end
+
+function panel:accept()
+  local entry = self:get_entry(0)
+  if not entry then
+    return
+  end
+
+  local bufnr = vim.uri_to_bufnr(panel_uri_to_doc_uri(self.panel_uri))
+
+  if not vim.api.nvim_buf_is_loaded(bufnr) then
+    vim.cmd("echoerr 'Buffer was closed'")
+    return
+  end
+
+  if vim.fn.getbufline(bufnr, entry.range.start.line + 1)[1] ~= self.state.line then
+    vim.cmd('echoerr "Buffer has changed since synthesizing solution"')
+    return
+  end
+
+  vim.api.nvim_buf_call(bufnr, function()
+    vim.lsp.util.apply_text_edits({
+      { range = entry.range, newText = entry.completionText },
+    }, 0, "utf-16")
+
+    if self.state.was_insert then
+      vim.cmd("startinsert!")
+    else
+      vim.cmd("normal! $")
+    end
+  end)
+
+  self:unlock():clear():lock()
+
+  vim.api.nvim_win_close(self.winid, true)
+end
+
+local function set_keymap(bufnr)
+  if panel.keymap.accept then
+    vim.keymap.set("n", panel.keymap.accept, mod.accept, {
+      buffer = bufnr,
+      desc = "[copilot] (panel) accept",
+      silent = true,
+    })
+  end
+
+  if panel.keymap.jump_prev then
+    vim.keymap.set("n", panel.keymap.jump_prev, mod.jump_prev, {
+      buffer = bufnr,
+      desc = "[copilot] (panel) jump prev",
+      silent = true,
+    })
+  end
+
+  if panel.keymap.jump_next then
+    vim.keymap.set("n", panel.keymap.jump_next, mod.jump_next, {
+      buffer = bufnr,
+      desc = "[copilot] (panel) jump next",
+      silent = true,
+    })
+  end
+
+  if panel.keymap.refresh then
+    vim.keymap.set("n", panel.keymap.refresh, mod.refresh, {
+      buffer = bufnr,
+      desc = "[copilot] (panel) refresh",
+      silent = true,
+    })
+  end
+end
+
+function panel:ensure_bufnr()
+  if self.bufnr and vim.api.nvim_buf_is_valid(self.bufnr) then
+    return
+  end
+
+  self.bufnr = vim.api.nvim_create_buf(false, true)
+
+  for name, value in pairs({
+    bufhidden = "hide",
+    buflisted = false,
+    buftype = "nofile",
+    filetype = "markdown",
+    modifiable = false,
+    readonly = true,
+    swapfile = false,
+    undolevels = 0,
+  }) do
+    vim.api.nvim_buf_set_option(self.bufnr, name, value)
+  end
+
+  set_keymap(self.bufnr)
+end
+
+function panel:ensure_winid()
+  if self.winid and vim.api.nvim_win_is_valid(self.winid) then
+    return
+  end
+
+  local height = math.floor(vim.api.nvim_win_get_height(0) * 0.4)
+
+  self.winid = vim.api.nvim_win_call(0, function()
+    vim.cmd(tostring(height) .. "split")
+    return vim.api.nvim_get_current_win()
+  end)
+
+  for name, value in pairs({
+    fcs = "eob: ",
+    list = false,
+    number = false,
+    numberwidth = 1,
+    relativenumber = false,
+    signcolumn = "no",
+  }) do
+    vim.api.nvim_win_set_option(self.winid, name, value)
+  end
```

Comment:
Add window conceal settings so markdown code block delimiters are hidden
```suggestion
  vim.schedule(function ()
    for name, value in pairs({
      fcs = "eob: ",
      list = false,
      number = false,
      numberwidth = 1,
      relativenumber = false,
      signcolumn = "no",
      conceallevel = 2,
      concealcursor = "nc"
    }) do
      vim.api.nvim_win_set_option(self.winid, name, value)
    end
  end)
```